web/react/components/navbar_dropdown.jsx

```diff
@@ -223,6 +233,18 @@ export default class NavbarDropdown extends React.Component {
                                 {'Report a Problem'}
                             </a>
                         </li>
+                        <li>
+                            <a
+                                href='#'
+                                onClick={this.handleAboutModal}
+                            >
+                                {'About Mattermost'}
+                            </a>
+                        </li>
+                        <AboutBuildModal
+                            show={this.state.showAboutModal}
+                            onModalDismissed={this.aboutModalDismissed}
+                        />
```

Comment:
Yup! It should be ok to put the modal right there, especially since moving it out would require wrapping the `ul` and the modal in a `span` or `div`